src/k8s-engine/kim.ts

```diff
@@ -125,20 +129,42 @@ class Kim extends EventEmitter {
     return this._isReady;
   }
 
-  async runCommand(args: string[], sendNotifications = true): Promise<childResultType> {
-    const child = spawn(resources.executable('kim'), args);
+  async runKimCommand(args: string[], sendNotifications = true): Promise<childResultType> {
+    return await this.runCommand(spawn(resources.executable('kim'), args), args[0], sendNotifications);
+  }
+
+  async runTrivyCommand(args: string[], sendNotifications = true): Promise<childResultType> {
+    let child: ChildProcess;
+    const kimCommand = args[0];
+
+    if (os.platform().startsWith('win')) {
+      args = ['-d', 'rancher-desktop', 'trivy'].concat(args);
+      child = spawn('wsl', args);
+    } else if (os.platform().startsWith('darwin')) {
+      const limaBackend = this.k8sManager as LimaBackend;
+
+      args = ['trivy'].concat(args);
+      child = limaBackend.limaSpawn(args);
+    } else {
+      throw new Error(`Don't know how to run trivy on platform ${ os.platform() }`);
+    }
+
+    return await this.runCommand(child, kimCommand, sendNotifications);
+  }
+
+  async runCommand(child: ChildProcess, kimCommand: string, sendNotifications: boolean): Promise<childResultType> {
```

Comment:
```suggestion
  async runCommand(child: ChildProcess, command: string, sendNotifications: boolean): Promise<childResultType> {
```
(Because we run pass the trivy command to this function too)

I guess we don't actually _run_ a command here, we just consume the output of the child process?

Reply:
Renamed the function to `processChildOutput` and the arg to `subcommandName`.

I found that by separating the all to `child_process.spawn` from the event processing we get more reuse and flexibility.